apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/state/GlobalVariables.java

```diff
@@ -51,7 +53,7 @@ public class GlobalVariables {
      */
     public static <T> T get(Class<?> adviceClass, String key, T defaultValue) {
         key = adviceClass.getName() + "." + key;
-        if (defaultValue.getClass().getClassLoader() != null && !defaultValue.getClass().getName().startsWith("co.elastic.apm.agent")) {
+        if (InternalUtil.getClassLoader(defaultValue.getClass(), false) == null && !defaultValue.getClass().getName().startsWith("co.elastic.apm.agent")) {
```

Comment:
`InternalUtil.getClassLoader(defaultValue.getClass(), false)` currently can't return `null`

Reply:
good catch, and also I did introduce a bug here, thus adding a test is required.